server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMessagesCreationProcessor.java

```diff
@@ -290,21 +296,41 @@ private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSe
         return new ValueWithId.MessageWithId(entry.getCreationId(), jmapMessage);
     }
 
-    private void assertUserIsSender(MailboxSession session, Optional<DraftEmailer> from) throws MailboxSendingNotAllowedException {
+    @VisibleForTesting
+    void assertUserIsSender(MailboxSession session, Optional<DraftEmailer> from) throws MailboxSendingNotAllowedException {
         if (!from.flatMap(DraftEmailer::getEmail)
-                .filter(email -> session.getUser().equals(Username.of(email)))
+                .filter(email -> emailBelongToTheConnectedUser(session, email))
                 .isPresent()) {
             String allowedSender = session.getUser().asString();
             throw new MailboxSendingNotAllowedException(allowedSender);
         }
     }
 
+    private boolean emailBelongToTheConnectedUser(MailboxSession session, String email) {
+        try {
+            return session.getUser().equals(Username.of(email)) || emailIsAnAliasOfTheConnectedUser(session, email);
+        } catch (RecipientRewriteTableException | RecipientRewriteTable.ErrorMappingException e) {
+            return false;
+        }
+    }
+
+    private boolean emailIsAnAliasOfTheConnectedUser(MailboxSession session, String email) throws RecipientRewriteTable.ErrorMappingException, RecipientRewriteTableException {
+        Username userFromMail = Username.of(email);
+        return session.getUser().getDomainPart().isPresent() &&
+            recipientRewriteTable.getResolvedMappings(userFromMail.getLocalPart(), userFromMail.getDomainPart().get())
+                .asStream().filter(mapping -> mapping.asMailAddress().isPresent())
+                .map(mapping -> mapping.asMailAddress().map(MailAddress::asString))
```

Comment:
The inner map is not necessary here?

Reply:
Yes, because the mapping `asString` method include the mapping type as a prefix.

Reply:
I meant more something like this:
```
.map(Mapping::asMailAddress)
.map(MailAddress::asString)
```
?